olp-cpp-sdk-core/tests/thread/TaskContinuationTest.cpp

```diff
@@ -210,24 +171,217 @@ TEST_F(TaskContinuationFixture, CallCancel) {
                 []() { return olp::client::CancellationToken(); },
                 []() { return olp::client::CancellationToken(); });
           })
-          .Finally([&promise,
-                    &result](olp::client::ApiResponse<olp::client::HttpResponse,
-                                                      olp::client::ApiError>
-                                 response) {
-            result = response.GetResult();
-            promise.set_value();
-          });
+          .Finally(
+              [&promise](ResponseType<olp::client::HttpResponse> response) {
+                promise.set_value(response.GetError());
+              });
 
   cont.Run();
-  std::this_thread::sleep_for(
-      std::chrono::milliseconds(50));  // simulate some delay
 
   cont.CancelToken().Cancel();
 
-  EXPECT_EQ(future.wait_for(std::chrono::milliseconds(1000)),
-            std::future_status::ready);
+  auto result = future.get();
```

Comment:
Please use the approach described above.

Reply:
done